docs/internals/optimizer.rst

```diff
@@ -1096,6 +1097,46 @@ The step LiteralRematerialiser is not required for correctness. It helps deal wi
 ``function f(x) -> y { revert(y, y} }`` where the literal ``y`` will be replaced by its value ``0``,
 allowing us to rewrite the function.
 
+.. _unused-store-eliminator:
+
+UnusedStoreEliminator
+^^^^^^^^^^^^^^^^^^^^^
+
+Optimizer component that removes ``sstore`` and ``mstore`` statements if conditions are met for said removal.
+In case of ``sstore``, if all code paths lead to a revert, or if a previous ``sstore`` is overwritten,
+the statement will be removed. For example, the following code
```

Comment:
Maybe it's because I wrote it, but it makes sense to me, especially with the example straight after it. I find your suggestion equally understandable, so am not against changing it.